test/Microsoft.Health.Fhir.Shared.Tests.E2E.Common/TestAuthenticationHttpMessageHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Common
+{
+    public class TestAuthenticationHttpMessageHandler : DelegatingHandler
```

Comment:
should this be in shared componnet? Seems like this would needed anywhere that sets the token right?

Reply:
This could be in a shared component but today it's only used in two audit tests as a way to override the default implementation in tests.

Also, I don't think the code should be in the `Microsoft.Health.Client` package so it would require a different package. I think the code is simple and one off enough today to not move yet.